### DISCUSSION TITLE
Decision Tables with Arrays of Objects

### DISCUSSION THREAD

**Discussion by rob-straughan** [gorules/zen#132] [Q&A] [answered]
Hi,

I'm currently doing a feasibility assessment of GoRules for one of our projects, and I have a question regarding using arrays in decision tables.

Say I have an input that looks like the following:

```
{
  objects: [
    {
      type: "type1",
      name: "name1"
    },
    {
      type: "type2",
      name: "name2"
    }
  ]
}
```

If I setup a decision table with a Collect output, what is the syntax I would use to get the answers for both objects?

I can easily get this to work for one element, by specifying the array index in the column, but that requires me to know how many elements are in the array ahead of time.  I have tried to use the # symbol to indicate a wildcard but that does not appear to be valid in the headers.

Screenshot and sample rule file attached.  Would appreciate any help, not sure if I'm just misreading the documentation.

Regards,
Rob.

![Screenshot 2024-03-14 122408](https://github.com/gorules/zen/assets/93327813/8f121d03-7138-45d7-9dbe-cb816abd66c2)
[test.json](https://github.com/gorules/zen/files/14602183/test.json)

**Comment by ivanmiletic** [comment, marked as answer]
Hi @rob-straughan at the moment we do not support re-mapping of the incoming array, this is something we will try to support soon.

**Reply by rob-straughan** [reply]
Thanks for that, for the time being I've worked around the issue by changing the way I structure the data, which has actually resulted in a simpler model anyway.  The feature would definitely be useful though.

Regards,
Rob.
